lib/Immediate/Immediate.cpp

```diff
@@ -293,6 +294,38 @@ int swift::RunImmediately(CompilerInstance &CI, const ProcessCmdLine &CmdLine,
   if (Context.hadError())
     return -1;
 
+  // Load libSwiftCore to setup process arguments.
+  //
+  // This must be done here, before any library loading has been done, to avoid
+  // racing with the static initializers in the standard library and user code.
+  auto stdlib = loadSwiftRuntime(Context.SearchPathOpts.RuntimeLibraryPath);
+  if (!stdlib) {
+    CI.getDiags().diagnose(SourceLoc(),
+                           diag::error_immediate_mode_missing_stdlib);
+    return -1;
+  }
+
+  // Setup interpreted process arguments.
+  using ArgOverride = void (*)(char **, int);
+  auto emplaceProcessArgs
+          = (ArgOverride)dlsym(stdlib, "_swift_stdlib_overrideUnsafeArgvArgc");
+  if (dlerror()) {
+    return -1;
+  }
+  char **argBuf = new char *[CmdLine.size()];
```

Comment:
Hm. Can't we just point to the C strings inside CmdLine? (We still have to make the new array, but we don't have to allocate the inner strings or strcpy.)

I'd also suggest using `SmallVector<const char *, 32>`, which lets the arguments stay on the stack if there aren't more than 32 of them.